outlook/src/taskpane/components/Log/Logger.tsx

```diff
@@ -27,21 +34,123 @@ class Logger extends React.Component<LoggerProps, LoggerState> {
         };
     }
 
-    private logRequest = (event) => {
+    private fetchAttachmentContent(attachment, index): Promise<any> {
+        return new Promise<any>((resolve) => {
+            if (attachment.size > SIZE_THRESHOLD_SINGLE_ELEMENT * 1024 * 1024) {
+                resolve({
+                    name: attachment.name,
+                    inline: attachment.isInline && attachment.contentType.indexOf("image") >= 0,
+                    oversize: true,
+                    index: index,
+                });
+            }
+            Office.context.mailbox.item.getAttachmentContentAsync(attachment.id, asyncResult => {
+                resolve({
+                    name: attachment.name,
+                    content: asyncResult.value.content,
+                    inline: attachment.isInline && attachment.contentType.indexOf("image") >= 0,
+                    oversize: false,
+                    index: index,
+                });
+            });
+        });
+    };
+
+    private logRequest = async (event): Promise<any> => {
         event.stopPropagation();
 
         this.setState({logged: 1});
-        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, (result) => {
+        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, async (result) => {
             const msgHeader = '<div>From: ' + Office.context.mailbox.item.sender.emailAddress + '</div><br/>';
-            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/user/crm/optimize/mail_client_extension.html" target="_blank">Outlook Inbox</a></div>';
+            const msgFooter = '<br/><div class="text-muted font-italic">Logged from <a href="https://www.odoo.com/documentation/14.0/applications/sales/crm/optimize/outlook_extension.html" target="_blank">Outlook Inbox</a></div>';
             const body = result.value.split('<div id="x_appendonsend"></div>')[0]; // Remove the history and only log the most recent message.
             const message = msgHeader + body + msgFooter;
-
+            const doc = new DOMParser().parseFromString(message, "text/html");
+            const officeAttachmentDetails = Office.context.mailbox.item.attachments;
+            let totalSize = 0;
+            const promises: any[] = [];
             const requestJson = {
                 res_id: this.props.resId,
                 model: this.props.model,
-                message: message
+                message: message,
+                attachments: [],
             }
+
+            //check if attachment size is bigger then the threshold
+            officeAttachmentDetails.forEach(officeAttachment => {
+                totalSize += officeAttachment.size
+            });
+
+            if (totalSize > SIZE_THRESHOLD_TOTAL * 1024 * 1024) {
+                const warningMessage = _t("Warning: Could not fetch attachments," +
```

Comment:
Note that in gmail we have "Attachments could not be logged in Odoo because their total size exceeded the allowed maximum." sentence. We should probably use the same, isn't it ?

Reply:
Indeed, sentence changed!